nagra/select.py

```diff
@@ -286,17 +286,28 @@ def stm(self):
         )
         return stm()
 
-    def to_polars(self, *args) -> "LazyFrame":
+    def to_polars(self, *args, schema_overrides: dict | None = None) -> "LazyFrame":
+        """
+        Execute the query with given args and return a polars
+        LazyFrame. Optionally, `schema_overrides` can be provided to
+        override the column types inferred from the database schema
+        and the query operations.
+
+        Only supported with PostgreSQL.
+        """
         assert self.trn.flavor != "sqlite", "Polars is only supported with Postgresql"
         import polars
 
         schema = self.dtypes(with_optional=False)
         cursor = self.execute(*args)
-        columns = [c for c, _ in schema]
-        df = polars.LazyFrame(cursor, schema=columns)
+        pl_schema = dict(schema)
         if self._aliases:
             mapping = dict(zip((n for n, _ in schema), self._aliases))
```

Comment:
Since we only loop on on those element we don't need to have a dict as intermediate result
```suggestion
            mapping = zip((n for n, _ in schema), self._aliases)
```